### PR TITLE
Fix 273: Auto detect native image

### PR DESCRIPTION
**Pull request by anthonydahanne**
## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: boot/native_image.go [hunk 1]
```diff
@@ -66,15 +65,6 @@ func (n NativeImageClasspath) Contribute(layer libcnb.Layer) (libcnb.Layer, erro
 			string(filepath.ListSeparator),
 			strings.Join(cp, string(filepath.ListSeparator)),
 		)
-
```
Comment (pivotal-david-osullivan) [inline boot/native_image.go:69]: As discussed @anthonydahanne, this should not be removed as it is not related to auto-detecting native-image, it is actually part of the functionality that enables Spring Boot 3.0+'s native support - Boot produces a Jar with the `argfile` file present and this buildpack detects this, and sets the env var `BP_NATIVE_IMAGE_BUILD_ARGUMENTS_FILE` for the native-image buildpack to use when constructing the native-image command

File: boot/native_image_test.go [hunk 2]
```diff
@@ -116,24 +116,4 @@ func testNativeImage(t *testing.T, context spec.G, it spec.S) {
 			Expect(layer.LayerTypes.Launch).To(BeFalse())
 		})
 	})
-	context("Boot @argfile is found", func() {
```
Comment (pivotal-david-osullivan) [inline boot/native_image_test.go:119]: As above, since this tests the same functionality

**Review by pivotal-david-osullivan — CHANGES_REQUESTED** [review]

File: boot/native_image_test.go (hunk 2 above)
Comment (anthonydahanne) [reply boot/native_image_test.go]: re instated!

**Comment by anthonydahanne** [comment]
hello @pivotal-david-osullivan , @dmikusa !
I've added some `sample` tests to assert that my changes, published through a [composite](https://github.com/anthonydahanne/java-native-image/commit/15501c7e91978faf947689a9cd57d75d4bbea6a0) and a [builder](https://github.com/anthonydahanne/tiny-builder/commit/9a63a7d073b41b277c442abae30185bd3d0b28c2), did not harm existing use cases; for [Spring Boot 3.0](https://github.com/paketo-buildpacks/samples/pull/563) builds, as well as [Spring Boot 3.1](https://github.com/paketo-buildpacks/samples/pull/564) - please disregarde the java node builds, they can't run on tiny builder.

I invite you to check those temporary samples out and... review one last time this PR!

Thanks a lot!

File: README.md [hunk 3]
```diff
@@ -25,6 +25,8 @@ The buildpack will do the following:
       * Contributes application slices as defined by the layer's index
     * If the application is a reactive web application
       * Configures `$BPL_JVM_THREAD_COUNT` to 50
+* If `<APPLICATION_ROOT>/META-INF/MANIFEST.MF` contains a `Spring-Boot-Native-Processed` entry OR if `$BP_MAVEN_ACTIVE_PROFILES` contains the `native` profile:
+  * A build plan entry is provided, `native-image-application`, which can be required by `native-image` to automatically trigger a native image build
```
Comment (pivotal-david-osullivan) [inline README.md:29]:
```suggestion
  * A build plan entry is provided, `native-image-application`, which can be required by the `native-image` [buildpack](https://github.com/paketo-buildpacks/native-image) to automatically trigger a native image build
```

**Review by pivotal-david-osullivan — APPROVED** [review]

**Pull request merged**